M2/Macaulay2/d/stdio.d

```diff
@@ -656,59 +656,37 @@ export present(x:string):string := (
      fixesneeded := 0;
      foreach cc in x do (
 	  c := cc; 
-	  if (c == '\0' ||
-	      c == '\"' ||
+	  if (c == '\"' ||
 	      c == '\\' ||
-	      c == '\a' ||
 	      c == '\b' ||
-	      c == '\e' ||
 	      c == '\f' ||
 	      c == '\n' ||
 	      c == '\r' ||
-	      c == '\t' ||
-	      c == '\v' )
-	  then fixesneeded = fixesneeded + 1 
+	      c == '\t' )
+	  then fixesneeded = fixesneeded + 1
+	  else if 0 <= c && c < 32 then fixesneeded = fixesneeded + 5
 	  );
      if fixesneeded != 0 then (
 	  new string len length(x)+fixesneeded do foreach cc in x do (
 	       c := cc;
-	       if      c == '\0' then (provide '\\'; provide '0')
-	       else if c == '\a' then (provide '\\'; provide 'a')
-	       else if c == '\b' then (provide '\\'; provide 'b')
-	       else if c == '\e' then (provide '\\'; provide 'e')
+	       -- control characters understood by JSON
+	       if      c == '\b' then (provide '\\'; provide 'b')
 	       else if c == '\f' then (provide '\\'; provide 'f')
 	       else if c == '\n' then (provide '\\'; provide 'n')
 	       else if c == '\r' then (provide '\\'; provide 'r')
 	       else if c == '\t' then (provide '\\'; provide 't')
-	       else if c == '\v' then (provide '\\'; provide 'v')
+	       else if 0 <= c && c < 32
+	       then ( -- escape all other control characters as \u00xy
```

Comment:
This is probably fine, but you should probably add some tests or otherwise document the behavior somewhere.

Reply:
Done!